libraries/chain/abi_serializer.cpp

```diff
@@ -32,15 +32,32 @@ namespace eosio { namespace chain {
    template <typename T>
    auto pack_unpack() {
       return std::make_pair<abi_serializer::unpack_function, abi_serializer::pack_function>(
-         []( fc::datastream<const char*>& stream, bool is_array, bool is_optional) -> fc::variant  {
-            if( is_array )
+         []( fc::datastream<const char*>& stream, bool is_array, bool is_optional, int fixed_array_size) -> fc::variant  {
+            if (fixed_array_size > 0) {
+               EOS_ASSERT(fixed_array_size <= MAX_NUM_ARRAY_ELEMENTS, unpack_exception, "invalid fixed array size ${s}", ("s", fixed_array_size));
```

Comment:
Do we know what pack will do when called with a very deep array with each level of size MAX_NUM_ARRAY_ELEMENTS?

Reply:
I see. I guess need to make the vector grows at the time of unpacking, so that the vector size is roughly capped by the size of stream.